python/cuda_parallel/tests/test_segmented_reduce_api.py

```diff
@@ -105,3 +105,133 @@ def scale(row_id):
     expected = cp.asarray(np.sum(mat, axis=-1))
     assert cp.all(d_output == expected)
     # example-end segmented-reduce-columnwise-total
+
+
+def test_segmented_reduce_for_columnwise_max():
+    # example-begin segmented-reduce-columnwise-maximum
+    import cupy as cp
+    import numpy as np
+
+    import cuda.parallel.experimental.algorithms as algorithms
+    import cuda.parallel.experimental.iterators as iterators
+    from cuda.parallel.experimental.iterators._strided import make_ndarray_iterator
+
+    def binary_op(a, b):
+        return max(a, b)
+
+    n_rows, n_cols = 123456, 78
+    rng = np.random.default_rng()
```

Comment:
Maybe we can do this on device directly? Save a few copies
```suggestion
    rng = cp.random.default_rng()
```